### ISSUE TITLE
Become contributors

### ISSUE THREAD

**Issue by kamalmarhubi** [nix-rust/nix#554]
@asomers @Susurrus this one is for you! You've both been doing a fair amount of thoughtful contribution lately. As you've probably see, the current maintainers do work on the project in spurts. Having a couple more people will hopefully give us a more consistent cadence over all. What do you think?

cc @nix-rust/nix-maintainers

**Comment by asomers** [comment]
Does it come with a two week training session at a corporate retreat in Tahiti?

Joking aside, ... maybe.  What does it entail to be a maintainer of nix?  I've never maintained a single github project until earlier this week.

**Comment by Susurrus** [comment]
I agree, we're going to need a two-week Tahiti trip to really sort out the details!

I'm interested in being a maintainer, but I agree with @asomers I would like to know a little bit about what you guys are hoping for in terms of recruiting additional maintainers. Are you guys on IRC together ever, I'd be happy to jump on and I can meet the existing team and get a sense of your guys plan for the library and everything. I'm assuming we'd all be on the same page with this, but it'd be great to have a more formal introduction to the team and the project before I sign on the dotted line so to speak.

**Comment by Susurrus** [comment]
@kamalmarhubi Are you guys on IRC or anything? Would love to talk to you and maybe some of the other maintainers.

**Comment by fiveop** [comment]
So far, almost all communication I had with @posborne and @kamalmarhubi happend in issue and PR threads.

**Comment by Susurrus** [comment]
What about starting a gitter for nix? I tried to jump on yesterday to see if there is one, but there isn't. The `gtk-rs` organization has one and it's pretty helpful.

**Comment by Susurrus** [comment]
But I'll become a maintainer. Would definitely love to help get this CI stuff sorted out.

**Comment by kamalmarhubi** [comment]
Hi! Sorry for the delay, was on vacation. Re what it entails, I think roughly: responding to issues and code reviews, engaging in discussion of project direction and conventions, helping keep the project ticking—CI, releases, and so on.

**Comment by kamalmarhubi** [comment]
As for synchronous communications: we have an IRC channel that nobody responds to. I'm a bit wary of having another medium with an expectation of quick responses and failing to actually do that—it's kind of a crappy experience for whoever's looking for help. But I'm not against trying it out!

**Comment by Susurrus** [comment]
That's fine by me. We can keep the current async comms going unless we find something better. Let me know if there's anything else you want from me for this.

**Comment by posborne** [comment]
Yeah, I wouldn't be opposed to moving to gitter from IRC.  I'm not great at having the chat client up all the time and the notification from gitter can be handy.

**Comment by Susurrus** [comment]
What's the IRC channel? That should definitely be mentioned in the README if there is a dedicated one (the same applies to gitter if that's the direction this is going).

**Comment by posborne** [comment]
> What's the IRC channel? 

`#irc` on irc.mozilla.org.  I haven't been on in ages.  I agree that there isn't much point in having an IRC/Gitter channel if it isn't mentioned.

**Comment by Susurrus** [comment]
@posborne @fiveop @kamalmarhubi You guys still planning to move ahead with this? I'm willing to become a maintainer. I see the top priorities as:
  1. Get `trust` working for testing `nix` across all supported platforms
  2. Work on RFCs (if that is still the desired approach) to resolve issues like `Duration`/`TimeValLike`/etc. and if the return value of functions should be `String`.
  3. Work through the PR backlog and at least continue triaging things.

**Comment by posborne** [comment]
I don't think I have heard any objections, so I think we should move forward.  @kamalmarhubi or @carllerche have the permissions to make this official.

**Comment by kamalmarhubi** [comment]
I'm sorry for the delay! I've invited @Susurrus to the org and maintainers group.

@asomers what say you?

**Comment by kamalmarhubi** [comment]
@Susurrus I enjoy your list!

**Comment by asomers** [comment]
I'm afraid that my time is too limited to be a general maintainer ATM.  But I'm happy to maintain select areas, like AIO and FreeBSD.  Feel free to assign bugs to me.

**Comment by asomers** [comment]
I'm afraid that my time is too limited to be a general maintainer ATM.  But I'm happy to maintain select areas, like AIO and FreeBSD.  Feel free to assign bugs to me.

**Comment by asomers** [comment]
Let me clarify: please do make me a maintainer.  I don't think I can be assigned bugs otherwise, nor can I review PRs; only comment on them.

**Comment by kamalmarhubi** [comment]
@Susurrus @asomers one tiny thing: we use @homu to handle merging, so when accepting a PR write something like
> @homu r+

This will run the tests on the merged result, and automatically merge when they pass. Benefits of us not having to poll the CI to know when to merge, and also testing the result of the merge instead of the branch being merged in.

**Comment by kamalmarhubi** [comment]
You can undo an approval with
> @homu r-
and get the tests retried with
> @homu retry
